great_expectations/experimental/datasources/data_asset/data_connector/file_path_data_connector.py

```diff
@@ -0,0 +1,303 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import TYPE_CHECKING, Callable, List, Optional
+
+from great_expectations.core.batch_spec import BatchSpec, PathBatchSpec
+from great_expectations.datasource.data_connector.batch_filter import (
+    BatchFilter,
+    build_batch_filter,
+)
+from great_expectations.datasource.data_connector.util import (
+    map_batch_definition_to_data_reference_string_using_regex,
+)
+from great_expectations.experimental.datasources.data_asset.data_connector.data_connector import (
+    DataConnector,
+)
+from great_expectations.experimental.datasources.data_asset.data_connector.regex_parser import (
+    RegExParser,
+)
+
+# TODO: <Alex>ALEX_INCLUDE_SORTERS_FUNCTIONALITY_UNDER_PYDANTIC-MAKE_SURE_SORTER_CONFIGURATIONS_ARE_VALIDATED</Alex>
+# TODO: <Alex>ALEX</Alex>
+# from great_expectations.experimental.data_asset.data_connector.sorter import Sorter
+# TODO: <Alex>ALEX</Alex>
+from great_expectations.experimental.datasources.data_asset.data_connector.util import (
+    batch_definition_matches_batch_request,
+    map_data_reference_string_to_batch_definition_list_using_regex,
+)
+
+if TYPE_CHECKING:
+    from great_expectations.core.batch import BatchDefinition
+    from great_expectations.experimental.datasources.interfaces import BatchRequest
+
+
+logger = logging.getLogger(__name__)
+
+
+class FilePathDataConnector(DataConnector):
+    """The base class for Data Connectors designed to access filesystem-like data.
+
+    This can include traditional, disk-based filesystems or object stores such as S3, GCS, or Azure Blob Store.
+    # TODO: <Alex>ALEX_INCLUDE_SORTERS_FUNCTIONALITY_UNDER_PYDANTIC-MAKE_SURE_SORTER_CONFIGURATIONS_ARE_VALIDATED</Alex>
+    # TODO: <Alex>ALEX</Alex>
+    # This class supports a regular expression and sorters for filtering and sorting data references.
+    # TODO: <Alex>ALEX</Alex>
+
+    See the `DataConnector` base class for more information on the role of Data Connectors.
+
+    Note that `FilePathDataConnector` is not meant to be used on its own, but extended.
+
+    Args:
+        name: The name of the Data Connector.
+        regex: A regex pattern for filtering data references
+        # TODO: <Alex>ALEX_INCLUDE_SORTERS_FUNCTIONALITY_UNDER_PYDANTIC-MAKE_SURE_SORTER_CONFIGURATIONS_ARE_VALIDATED</Alex>
+        # TODO: <Alex>ALEX</Alex>
+        # sorters: A list of sorters for sorting data references.
+        # TODO: <Alex>ALEX</Alex>
+    """
+
+    def __init__(
+        self,
+        name: str,
+        datasource_name: str,
```

Comment:
consider references to types instead of labels

Reply:
@donaldheppner Please see the above reply on the subject of separation of concerns.  Thanks.

Reply:
This burdens the called with having to lookup something instead of having ready access to the reference. I also don't understand the use-case for having this data stored in the data connector. In ZEP, I'd think the functionality of the data connector should be limited to a few utility-like methods that datasources or assets use

Reply:
@donaldheppner Please my proposal above that would remove `datasource_name` from `DataConnector` as the instance variable and instead pass it down through one of the interface methods.  This would almost certainly require the deletion of the `DataConnector.self_check()` method, which you are advocating (with the trade-off of somewhat longer unit test code).